DropboxLike.Domain/Repositories/FileRepository.cs

```diff
@@ -47,127 +47,167 @@ public async Task<S3Response> UploadFileAsync(IFormFile file)
 
         await transferUtility.UploadAsync(uploadRequest);
 
-        var fileModel = new FileModel
+        var fileModel = new FileEntity
         {
           FileKey = uploadRequest.Key,
           FileName = file.FileName,
-          FileSize = (file.Length).ToString(),
+          FileSize = file.Length.ToString(),
           ContentType = file.ContentType,
-          TimeStamp = (DateTime.UtcNow).ToString()
+          TimeStamp = DateTime.UtcNow.ToString(CultureInfo.InvariantCulture)
         };
 
         _applicationDbContext.FileModels.Add(fileModel);
-        // TODO: Check if file already exist
         await _applicationDbContext.SaveChangesAsync();
-
-        _response.StatusCode = 200;
-        _response.Message = $"{file.Name} has been uploaded to s3 successfully";
+        
+        return OperationResult<object>.Success(new object(), HttpStatusCode.Created);
       }
     }
-    catch (AmazonS3Exception ex)
+    catch (AmazonS3Exception exception)
     {
-      _response.StatusCode = (int)ex.StatusCode;
-      _response.Message = ex.Message;
+      var message = $"{exception.StatusCode}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message, exception.StatusCode);
     }
-    catch (Exception ex)
+    catch (Exception exception)
     {
-      _response.StatusCode = 500;
-      _response.Message = ex.Message;
+      var message = $"{HttpStatusCode.InternalServerError}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message);
     }
-    return _response;
   }
 
-  public async Task<FileStreamResult> DownloadFileAsync(string fileId)
+  public async Task<OperationResult<File>> DownloadFileAsync(string fileId)
   {
-
-    // var results = await _applicationDbContext.FileModels
-    //     .FirstOrDefaultAsync(x => x.FileKey == fileId);
-
-    //var destinationFolderPath = $"/home/godfreyowidi/Downloads/TestsDowloads";
-
     try
     {
       var file = await _applicationDbContext.FileModels.FindAsync(fileId);
-
       if (file == null)
       {
         throw new FileNotFoundException("File not Found");
       }
 
-      var objectKey = file?.FileKey?.ToString();
-
-      ListObjectsV2Request listRequest = new ListObjectsV2Request
+      var listRequest = new ListObjectsV2Request
       {
         BucketName = _bucketName,
       };
-      ListObjectsV2Response listResponse = await _awsS3Client.ListObjectsV2Async(listRequest);
+      
+      // This code just ensure we retrieve only object names - not all metadata
+      // Caching of the retrieved names for 20 mins - server side caching.
+      var cache = new MemoryCache(new MemoryCacheOptions());
```

Comment:
Should this cache not be initialized as a singleton? If you reinitialize it on every call (every call creates a new instance of the repository and what's more, every call of DownloadFileAsync creates a new cache instance), the changes made to the cache in previous calls are lost. Did you test this manually? One way to verify that it works would be to monitor the response time for two identical calls: the response time of the second call should be way lower.